compiler/modules/front/token.bal

```diff
@@ -219,6 +221,35 @@ class Tokenizer {
                 }
                 return [IDENTIFIER, ident];
             }
+            else if ch == "0" {
+                ch = self.getc();
+                if ch == "x" || ch == "X" {
+                    string hex = "";
+                    while true {
+                        ch = self.getc();
+                        if ch is () {
+                            break;
+                        }
+                        else if !HEX_DIGIT.includes(ch) {
+                            self.ungetc(ch);
+                            break;
+                        }
+                        else {
+                            hex += ch;
+                        }
+                    }
+                    return [HEX_INT_LITERAL, hex];
```

Comment:
should I validate length here?, currently validated in the parser.

Reply:
No: leave it to the parser. Depends on whether we are doing int or decimal.

Reply:
Oh, I meant the `hex.lenght() == 0` case. Hope answer is still the same.

Reply:
I would catch length of zero in the tokenizer: it's not a valid token.